### ISSUE TITLE
Empty tracing console when using Loadbalancer with SSL-Termination

### ISSUE THREAD

**Issue by thomschke** [kiali/kiali#459]
**Describe the bug**
If Kiali is running behind a loadbalancer with SSL-Termination
then the tracing console is empty.

**Versions used**
Kiali: istio-release-1.0
Istio: 1.0
Kubernetes flavour and version: 1.10.6-gke.1

**To Reproduce**
1. Define an gateway (https) and a virtualservice for kiali 
2. Call the tracing console

**Details**
Before opening tracing console, there is a prefetch to `/api/jaeger` which response with an url for the  jaeger reverseproxy.

There are two problems:

a) The url has a wrong scheme:

https://github.com/kiali/kiali/blob/23643f7f6a099041692205ba5affa7b3a91cabdd/handlers/jaeger.go#L34-L37

You can query the http-header `x-forwarded-proto` in this context, isn't it?

b) The url uses an seperate fixed port `32439` (instead path)

https://github.com/kiali/kiali/blob/23643f7f6a099041692205ba5affa7b3a91cabdd/handlers/jaeger.go#L40-L42

So I have to open an additional port `32439` on the gateway for this behavior. Why do not you use a different path?

**Comment by lucasponce** [comment]
Related:
https://groups.google.com/forum/#!topic/kiali-dev/6D8U_gIh-j4

**Comment by thomschke** [comment]
There is already a property JAEGER_URL in istio-kiali-chart, but so far without any function:

https://github.com/istio/istio/blob/7904b4d58844582f4e87189020938b840bc0f378/install/kubernetes/helm/istio/charts/kiali/templates/deployment.yaml#L63-L66

I have no problem to publish the URLs for externalservices (jaeger/grafana), because you can protect this services with mTLS [(see)](https://istio.io/docs/tasks/traffic-management/secure-ingress/#configure-a-mutual-tls-ingress-gateway).

But the wrong scheme (topic a) is a showstopper for me in the moment. Can you fix this by quering `x-forwarded-proto` ?

**Comment by lucasponce** [comment]
Thanks @thomschke for the PR and the feedback.
Right now as there are several ways we can expose jaeger-query service, we are thinking to rely on the configuration and try to not infer the method jaeger-query was exposed (i.e. in OpenShift you can use a route, or use an Ingress, etc).
So we think that it could be more simple just to read the url from config as users can have different ways to expose this (also when introducing SSL).

What do you think ? We would like to hear also feedback.
Implementation is not really a problem, indeed, it is a minor change, but we want to address the most easy/useful way.

Note: we are also discussing similar/related topic on PR https://github.com/kiali/kiali/pull/460

**Comment by thomschke** [comment]
Yes, I would prefere the config way too in the same way as GRAFANA_URL.
https://github.com/kiali/kiali/blob/e7dbfbd8cb2d916e1b5d456252bb98de2cfcc88c/config/config.go#L36

**Comment by lucasponce** [comment]
cc @aljesusg related to #460

**Comment by aljesusg** [comment]
We'll have the configuration like GRAFANA, If you set a Jaeger URL configuration Kiali will provide you this URL, if not KIALI will find the URL in openshift/kubernetes

**Comment by thomschke** [comment]
It works. GJ !
